Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,334 @@
+
+                        // Create buffer
+                        byte[] buffer = new byte[(int)size];
+
+                        // Get input stream and the size of the original stream
+                        using (var inputStream = speechStream.GetInputStreamAt(0))
+                        {
+                            // Close the original speech stream to free up memory
+                            speechStream.Dispose();
+
+                            // Create a new data reader off the input stream
+                            using (var dataReader = new DataReader(inputStream))
+                            {
+                                // Load all bytes into the reader
+                                await dataReader.LoadAsync((uint)size);
+
+                                // Copy from reader into buffer
+                                dataReader.ReadBytes(buffer);
+                            }
+                        }
+
+                        // Convert raw WAV data into Unity audio data
+                        int sampleCount = 0;
+                        int frequency = 0;
+                        var unityData = ToUnityAudio(buffer, out sampleCount, out frequency);
+
+                        // The remainder must be done back on Unity's main thread
+                        UnityEngine.WSA.Application.InvokeOnAppThread(() =>
+                        {
+                            // Convert to an audio clip
+                            var clip = ToClip("Speech", unityData, sampleCount, frequency);
+
+                            // Set the source on the audio clip
+                            audioSource.clip = clip;
+
+                            // Play audio
+                            audioSource.Play();
+                        }, false);
+                    });
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogErrorFormat("Speech generation problem: \"{0}\"", ex.Message);
+                }
+            }
+            else
+            {
+                Debug.LogErrorFormat("Speech not initialized. \"{0}\"", text);
+            }
+        }
+        #endif
+
+        private void StartSpeech()
+        {
+            try
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("An AudioSource is required and should be assigned to 'Audio Source' in the inspector.");
+                }
+                else
+                { 
+                    #if WINDOWS_UWP
+                    synthesizer = new SpeechSynthesizer();
+                    #endif
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError("Could not start Speech Synthesis");
+                Debug.LogException(ex);
+            }
+        }
+
+        // MonoBehaviour Methods
+        void Start()
+        {
+            // Start speech
+            StartSpeech();
```

Comment:
These were the last few comments I had, so, as long as nobody else has anything else to say, I'm happy with the state of things.